### ISSUE TITLE
list-ops: add test to check if list was changed

### ISSUE THREAD

**Issue by tehsphinx** [exercism/go#1223]
I just ran across a solution to `listops` that does pass the tests but that I would see as very problematic in the real world. Using the `Append` function as an example:

```
func (l *IntList) Append(v IntList) IntList {
	result := []int(*l)
	for _, v := range []int(v) {
		result = append(result, v)
	}

	return result
}
```

After `Append` was called I would expect my list `l` to have changed. After all I did call a method on the list. But it hasn't.

1) My suggestion would be to add tests for this not only comparing the output of the `Append` function but also the list itself with the intended result.

```
func TestAppendMethod(t *testing.T) {
	for _, tt := range appendTestCases {
		got := tt.list.Append(tt.appendThis)
		if !reflect.DeepEqual(tt.want, got) {
			t.Fatalf("FAIL: %s: %q -- expected: %v, actual: %v", tt.property, tt.name, tt.want, got)
		} else if !reflect.DeepEqual(tt.want, tt.list) {
			t.Fatalf("FAIL: %s: %q checking list -- expected: %v, actual: %v",
				tt.property, tt.name, tt.want, tt.list)
		} else {
			t.Logf("PASS: %s: %s", tt.property, tt.name)
		}
	}
}
```

2) We could also change the tests altogether for the functions not to have a return on **any** of the functions, but to only ever check the list object itself against the desired state. The fact that each function returns a `*IntList` is nice for chaining but could be misleading/confusing in this exercise.

```
func TestAppendMethod(t *testing.T) {
	for _, tt := range appendTestCases {
		tt.list.Append(tt.appendThis)
		if !reflect.DeepEqual(tt.want, tt.list) {
			t.Fatalf("FAIL: %s: %q -- expected: %v, actual: %v",
				tt.property, tt.name, tt.want, tt.list)
		} else {
			t.Logf("PASS: %s: %s", tt.property, tt.name)
		}
	}
}
```

What are your thoughts?

**Comment by brugnara** [comment]
Hi @tehsphinx.

While I do understand your point, I am not sure that's the wanted behaviour here, nor in Go.

From the [A Tour of Go](https://tour.golang.org/moretypes/15):

```
The resulting value of append is a slice containing all the elements of the original slice plus the provided values.
```

We are almost used to do this way in Go, while it's quite different in JS, just as an example, where one can choose between `.push` and `.concat`.

- [concat function](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/concat)
- [push function](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/push)

In a nutshell I would expect an inplace append with no return at all, and no chaining option for the original slice either.

Maybe I'm wrong but also open to chat about this further 💪 .
This is my thought 😄
